aospy/data_loader.py

```diff
@@ -0,0 +1,168 @@
+                da = ds[name]
+                return da.rename({name: var.name})
+            except KeyError:
+                pass
+        raise KeyError('{0} not found in '
+                       'among names:{1} in {2}'.format(var, var.names, ds))
+
+    @staticmethod
+    def rename_grid_attrs(ds):
+        """Renames existing grid attributes to be consistent with
+        aospy conventions.  Does not compare to Model coordinates or
+        add missing coordinates from Model objects.  Grid attributes
+        are set as coordinates, such that they are carried by all
+        selected DataArrays with overlapping index dimensions.
+
+        Parameters
+        ----------
+        ds : Dataset
+
+        Returns
+        -------
+        renamed : Dataset
+            Dataset returned with coordinates consistent with aospy
+            conventions
+        """
+        # TODO: Separate coord setting and coord renaming.  Coord renaming
+        # should come before coord setting.  Set coords just before you select
+        # the variable.
+        for name_int, names_ext in GRID_ATTRS.items():
+            ds_coord_name = set(names_ext).intersection(set(ds.coords) |
+                                                        set(ds.data_vars))
+            if ds_coord_name:
+                ds.rename({ds_coord_name.pop(): name_int}, inplace=True)
+                # ds.set_coords(name_int, inplace=True)
+        return ds
+
+    @staticmethod
+    def set_grid_attrs_as_coords(ds):
+        """Sets available grid attributes as coordinates in a given Dataset.
+        Grid attributes are assumed to have their internal aospy names.
+
+        Parameters
+        ----------
+        ds : Dataset
+            Input data
+
+        Returns
+        -------
+        Dataset
+            Dataset with grid attributes set as coordinates
+        """
+        int_names = GRID_ATTRS.keys()
+        grid_attrs_in_ds = set(int_names).intersection(set(ds.coords) |
+                                                       set(ds.data_vars))
+        ds.set_coords(grid_attrs_in_ds, inplace=True)
+        return ds
+
+    def _generate_file_set(self, var, start_date, end_date, intvl_in):
+        raise NotImplementedError(
+            'All DataLoaders require a _generate_file_set method')
+
+
+class DictDataLoader(DataLoader):
+    """A data loader that corresponds with a dictionary mapping lists of files
+    to interval in tags.
+    """
+    def __init__(self, file_map={}, correction=None):
+        """Instantiates a new `DictDataLoader`
+
+        Parameters
+        ----------
+        file_map : dict
+            A dict mapping an input interval to a list of files
+        correction : function
+            An optional function to correct an aspect of input data before
+            subsetting in time
+        """
+        self.file_map = file_map
+        self.correction = correction
+
+    def _generate_file_set(self, var, start_date, end_date, intvl_in):
```

Comment:
Seems a bit clunky to have start_date and end_date as required but unused args that go before intvl_in.  Is there potentially a better call signature?  Tricky because the requirements/logic will vary across DataLoader implementations